src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/beanvalidation/BeanValidationQuickFix.java

```diff
@@ -1,82 +1,138 @@
  * 2. Removing static modifier from field or method
- * 
- * @author Leslie Dawson (lamminade)
  *
+ * @author Leslie Dawson (lamminade)
  */
-public class BeanValidationQuickFix {
+public class BeanValidationQuickFix implements IJavaCodeActionParticipant {
+
+    private static final Logger LOGGER = Logger.getLogger(BeanValidationQuickFix.class.getName());
+
+    @Override
+    public String getParticipantId() {
+        return BeanValidationQuickFix.class.getName();
+    }
 
     public List<? extends CodeAction> getCodeActions(JavaCodeActionContext context, Diagnostic diagnostic) {
         List<CodeAction> codeActions = new ArrayList<>();
-        removeConstraintAnnotations(diagnostic, context.copy(), codeActions);
+        getRemoveConstraintAnnotationsCodeActions(diagnostic, context.copy(), codeActions);
 
         if (diagnostic.getCode().getLeft().equals(BeanValidationConstants.DIAGNOSTIC_CODE_STATIC)) {
-            removeStaticModifier(diagnostic, context.copy(), codeActions);
+            getRemoveStaticModifierCodeActions(diagnostic, context.copy(), codeActions);
         }
         return codeActions;
     }
 
-    private void removeConstraintAnnotations(Diagnostic diagnostic, JavaCodeActionContext context, List<CodeAction> codeActions) {
+    @Override
+    public CodeAction resolveCodeAction(JavaCodeActionResolveContext context) {
+
+        final CodeAction toResolve = context.getUnresolved();
+        List<Diagnostic> diagnostics = toResolve.getDiagnostics();
+
+
+        String message = toResolve.getTitle();
+
+        if (message.equals(Messages.getMessage("RemoveStaticModifier"))) {
+            resolveStaticModifierCodeAction(context);
+            return toResolve;
+        }
+
+        resolveRemoveConstraintAnnotationsCodeAction(context, diagnostics.size() > 0 ? diagnostics.get(0) : null);
+
+        return toResolve;
+    }
+
+    private void getRemoveConstraintAnnotationsCodeActions(Diagnostic diagnostic, JavaCodeActionContext context, List<CodeAction> codeActions) {
+
+        final String annotationName = diagnostic.getData().toString().replace("\"", "");
+        final String name = Messages.getMessage("RemoveConstraintAnnotation", annotationName);
+        codeActions.add(JDTUtils.createCodeAction(context, diagnostic, name, getParticipantId()));
+    }
+
+    private void resolveRemoveConstraintAnnotationsCodeAction(JavaCodeActionResolveContext context, Diagnostic diagnostic) {
+
+        final CodeAction toResolve = context.getUnresolved();
         final PsiElement node = context.getCoveredNode();
         final PsiClass parentType = PsiTreeUtil.getParentOfType(node, PsiClass.class);
-        final PsiModifierListOwner modifierListOwner = PsiTreeUtil.getParentOfType(node, PsiModifierListOwner.class);
-
         final String annotationName = diagnostic.getData().toString().replace("\"", "");
+        final String name = Messages.getMessage("RemoveConstraintAnnotation", annotationName);
```

Comment:
I noticed the same messages are being retrieved/computed multiple times. Isn't this already available from the `CodeAction` object that you created?